[DataGrid] Rename classes according to new convention (#1872)
mui-org/material-ui-x: docs/pages/api-docs/data-grid/data-grid.md

| <span class="prop-name">cell</span> | <span class="prop-name">.MuiDataGrid-cell</span> | Styles applied to the cell element.|
|  | <span class="prop-name">.MuiDataGrid-columnHeaderCheckbox</span> | Styles applied to the header checkbox cell element.|
|  | <span class="prop-name">.MuiDataGrid-cellCheckbox</span> | Styles applied to the cell checkbox element.|
|  | <span class="prop-name">.MuiDataGrid-columnHeader--sortable</span> | Styles applied to the sortable header cell element.|
|  | <span class="prop-name">.MuiDataGrid-sortIcon</span> | Styles applied to the sort icon element.|
|  | <span class="prop-name">.MuiDataGrid-columnHeader--alignCenter</span> | Styles applied to the centered header cell element.|
|  | <span class="prop-name">.MuiDataGrid-columnHeader--alignRight</span> | Styles applied to the aligned right header cell element.|
|  | <span class="prop-name">.MuiDataGrid-columnHeaderTitle</span> | Styles applied to the header cell title element.|
|  | <span class="prop-name">.MuiDataGrid-columnSeparator</span> | Styles applied to the header cell separator element.|

> |  | <span class="prop-name">.MuiDataGrid-withColumnMenu</span> | Styles applied to the header cell that has a column menu.|

Reviewer: Not used anymore?

```suggestion
```

Author: A relic of the past I guess 😅

|  | <span class="prop-name">.MuiDataGrid-window</span> | Styles applied to the window element.|
|  | <span class="prop-name">.MuiDataGrid-viewport</span> | Styles applied to the viewport element.|
| <span class="prop-name">row</span> | <span class="prop-name">.MuiDataGrid-row</span> | Styles applied to the row element.|

> |  | <span class="prop-name">.Mui-selected</span> | Pseudo-class applied to the selected elements.|

Reviewer: ```suggestion
```

Author: Not sure why do we need to remove that one?

Reviewer: Since the core documents them. you can ignore my comment.

![image](https://user-images.githubusercontent.com/42154031/122047210-3dd45680-cdb6-11eb-8015-af2dc34a181e.png)

|  | <span class="prop-name">.MuiDataGrid-cell--withRenderer</span> | Styles applied to the customized cell element.|
|  | <span class="prop-name">.MuiDataGrid-withBorder</span> | Styles applied to the cell element that has right border displayed.|
|  | <span class="prop-name">.MuiDataGrid-cell--textLeft</span> | Styles applied to the aligned left cell element.|
|  | <span class="prop-name">.MuiDataGrid-cell--textRight</span> | Styles applied to the aligned right cell element.|
|  | <span class="prop-name">.MuiDataGrid-cell--textCenter</span> | Styles applied to the centered cell element.|

> |  | <span class="prop-name">.MuiGridFooterContainer</span> | Styles applied to the footer element.|

Reviewer: If it's standalone it doesn't need to appear here I think.

```suggestion
```

Author: Where would it appear then. I still think we need to keep them at least for now because there won't be documented anywhere. Also, the 'standalone' term is dangerous because we are implying that you would be able to use that component by itself where the truth is that its value is closely tied to the grid and although you could use it it wouldn't be beneficial.

Reviewer: As suggested in https://github.com/mui-org/material-ui-x/pull/1872#discussion_r651695266, there'll be no more standalone components so `.MuiGridFooterContainer` -> `.MuiDataDrid-footerContainer`.

|  | <span class="prop-name">.MuiDataGrid-viewport</span> | Styles applied to the viewport element.|
| <span class="prop-name">row</span> | <span class="prop-name">.MuiDataGrid-row</span> | Styles applied to the row element.|

> |  | <span class="prop-name">.Mui-selected</span> | Styles applied to the selected row element.|

Reviewer 2: There are no styles on this class name, it's a pseudo-class.
```suggestion
|  | <span class="prop-name">.Mui-selected</span> | Pseudo-class applied to the selected elements.|
```
